src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -441,6 +441,12 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
             break;
 #endif // TARGET_ARM64
 
+#ifdef SWIFT_SUPPORT
+        case GT_SWIFT_ERROR:
+            treeNode->SetRegNum(REG_SWIFT_ERROR);
```

Comment:
This should be implemented similarly to `genCodeForPhysReg` (in particular `genProduceReg` needs to be called)

Reply:
Not wholly relevant, but is there a reason why the platform-agnostic implementation of `genCodeForPhysReg` is in `codegenxarch.cpp`?

Reply:
I don't think so, seems like it could be in codegenlinear.cpp instead.

Reply:
I can move it in this PR if you think it makes sense to

Reply:
Well, the two implementations are identical, so I still think they could be unified in a common file. But probably best to leave it alone since RISCV64/LA64 might break from such a refactoring.